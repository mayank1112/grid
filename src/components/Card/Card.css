.contentCard {
    border: 1px solid orange;
    margin: 4px;
    border-radius: 5px;
    background: rgb(236, 131, 33);
    height: 80px;
    padding: 20px;
}

img {
    width: 50px;
    height: 50px;
}
