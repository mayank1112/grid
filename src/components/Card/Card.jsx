import axios from 'axios';
import { useEffect, useState } from 'react';
import Loader from '../Loader/Loader';
import './Card.css';

const Card = ({ card }) => {
    const [image, setImage] = useState(null);
    const [loading, setLoading] = useState(true);

    useEffect(() => {
        generateCard();
    }, []);

    const generateCard = async () => {
        setLoading(true);
        const { url } = card;
        const response = await axios.get(url);
        setImage(response.data.sprites.front_default);
        setLoading(false);
    };

    const { name } = card;
    return (
        <div className='contentCard'>
            <div>{!loading ? <img src={image} alt={name} /> : <Loader />}</div>
            <div>{name}</div>
        </div>
    );
};

Card.propTypes = {};

export default Card;
