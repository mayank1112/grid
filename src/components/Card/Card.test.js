import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Card from './Card';
import toJson from 'enzyme-to-json';
import axios from 'axios';
import { act } from 'react-dom/test-utils';

Enzyme.configure({ adapter: new Adapter() });

it('<Card> component should match snapshot', () => {
    const wrapper = shallow(<Card card={{ name: 'mockName', url: 'mockUrl' }} />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it('<Card> should call axios', async () => {
    const mockAxios = jest.spyOn(axios, 'get').mockResolvedValue(Promise.resolve({ data: { sprites: {} } }));
    await act(async () => {
        mount(<Card card={{ name: 'mockName', url: 'mockUrl' }} />);
    });
    expect(mockAxios).toHaveBeenCalledTimes(1);
});
