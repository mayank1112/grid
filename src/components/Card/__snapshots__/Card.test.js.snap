// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card> component should match snapshot 1`] = `
<div
  className="contentCard"
>
  <div>
    <Loader />
  </div>
  <div>
    mockName
  </div>
</div>
`;
